packages/foreman/foreman/foreman.spec

```diff
@@ -677,6 +677,20 @@ Configuration files for the Performance Co-Pilot integration
 %{_sysconfdir}/pcp/proc/%{name}-hotproc.conf
 %{_sharedstatedir}/pcp/config/pmlogconf/%{name}-hotproc
 
+%package profiling
+Summary: Adds rails-profiling extensions to foreman
+Requires: rubygem(rack-mini-profiler)
+Requires: rubygem(memory_profiler)
+Requires: rubygem(stackprof)
+
+%description profiling
+This automatically installs and enables profiling extensions to foreman.
+It is probably not what you want on a productive system.
+
+%files profiling
+%{_datadir}/%{name}/bundler.d/profiling.rb
+%{_datadir}/%{name}/config/initializers/rack_mini_profiler.rb
```

Comment:
I think by default this means the file is present in both `foreman` and `foreman-profiling`. Be sure to check that.

It may also be fine to include it in core if it gracefully falls back if the dependencies aren't present. Though there may be a security concern so I'm not strict on that. Just want to make sure you're doing what you intend to do.

Reply:
So what I can say from trying it out is that in 'productive`-mode, which means `Gemfile.In` is present, all available groups are loaded, regardless of it being optional or not.

When we use bundler, groups matter, but even though I do `bundle --with profiling`, the group would have to be required somewhere in code like here: https://github.com/theforeman/foreman/blob/fa79806973c6de08b6d265ed1ae99c0ff4f06bd6/config/application.rb#L59

Assuming we want to have it in `development`-environment and in `production`, if it is installed, then I would keep it in `bundler.d/profiling.rb` but add it to `development`-group.

> I think by default this means the file is present in both foreman and foreman-profiling. Be sure to check that.

AFAIK having the same filepath in multiple packages is a big NoNo in most packaging-frameworks and my first test suggests it is only in the `foreman-profiling` package after build.

Reply:
> So what I can say from trying it out is that in `productive`-mode, which means `Gemfile.In` is present, all available groups are loaded, regardless of it being optional or not.

That's due to [bundler_ext](https://github.com/bundlerext/bundler_ext), which we use on Red Hat. We don't on Debian. Perhaps this is a good excuse to drop bundler_ext usage. The original use case was that you could ignore gem dependencies in RPMs, for security patching. These days we just make sure the gem dependencies are correct.

> AFAIK having the same filepath in multiple packages is a big NoNo in most packaging-frameworks and my first test suggests it is only in the `foreman-profiling` package after build.

I know I've seen weirdness with this in the past, so just wanted to highlight that it should be checked.